packages/flame/lib/src/components/component_set.dart

```diff
@@ -133,6 +133,21 @@ class ComponentSet extends QueryableOrderedSet<Component> {
     _removeLater.addAll(this);
   }
 
+  /// Return sublist of all components of type [C].
+  ///
+  /// This is equivalent to `List.whereType<C>()`, except the result of this
+  /// operation is memoized for each distinct type [C]. The type [C] may be
+  /// "registered" in advance using `register<C>()`, or it may not, in which
+  /// case the type will be registered at the time of the first query. There
+  /// is no performance difference between these two approaches.
```

Comment:
The last part of this comment should be changed or removed since it isn't true.

Reply:
What I meant is that the total amount of CPU time spent on computing the cached list will be the same in both cases. Hence, neither of the approaches can be claimed to be faster or slower than the other.

Reply:
That isn't necessarily true either, since a tree with many different types of components will take longer to go through if the query isn't registered first.

Reply:
ok, I removed that sentence then.